metadatastore/commands.py

```diff
@@ -211,76 +616,53 @@ def insert_run_stop(run_start, time, uid=None, exit_status='success',
+    # look up the oid
+    runstart_oid = _RUNSTART_UID_to_OID_MAP[runstart_uid]
+    # create a reference field
+    rs_ref = DBRef('RunStart', runstart_oid)
+
+    run_stop = RunStop(run_start=rs_ref, reason=reason, time=time,
+                       uid=uid,
                        exit_status=exit_status, **custom)
 
-    run_stop.save(validate=True, write_concern={"w": 1})
+    run_stop = run_stop.save(validate=True, write_concern={"w": 1})
+    _cache_runstop(run_stop.to_mongo().to_dict())
     logger.debug("Inserted RunStop with uid %s referencing RunStart "
-                 " with uid %s", run_stop.uid, run_start.uid)
+                 " with uid %s", run_stop.uid, run_start['uid'])
 
     return uid
 
 
 @_ensure_connection
-def insert_beamline_config(config_params, time, uid=None):
-    """ Create a beamline_config  in metadatastore database backend
-
-    Parameters
-    ----------
-    config_params : dict
-        Name/value pairs that indicate beamline configuration
-        parameters during capturing of data
-    time : float
-        The date/time as found at the client side when the
-        beamline configuration is created.
-    uid : str, optional
-        Globally unique id string provided to metadatastore
-
-    Returns
-    -------
-    blc : BeamlineConfig
-        The document added to the collection
-    """
-    if uid is None:
-        uid = str(uuid.uuid4())
-    beamline_config = BeamlineConfig(config_params=config_params,
-                                     time=time,
-                                     uid=uid)
-    beamline_config.save(validate=True, write_concern={"w": 1})
-    logger.debug("Inserted BeamlineConfig with uid %s",
-                 beamline_config.uid)
-
-    return uid
-
-
-@_ensure_connection
-def insert_event_descriptor(run_start, data_keys, time, uid=None,
+def insert_descriptor(run_start, data_keys, time, uid,
                             custom=None):
-    """ Create an event_descriptor in metadatastore database backend
+    """Inesrt an EventDescriptor document in to  database.
 
     Parameters
     ----------
-    run_start : metadatastore.documents.Document or str
-        if Document:
-            The metadatastore RunStart document
-        if str:
-            uid of RunStart object to associate with this record
+    run_start : doc.Document or dict or str
+        The RunStart to get the RunStop for.  Can be either
+        a Document/dict with a 'uid' key or a uid string
     data_keys : dict
         Provides information about keys of the data dictionary in
-        an event will contain
+        an event will contain.  No key name may include '.'
     time : float
         The date/time as found at the client side when an event
         descriptor is created.
-    uid : str, optional
+    uid : str
         Globally unique id string provided to metadatastore
     custom : dict, optional
         Any additional information that data acquisition code/user wants
```

Comment:
Note that this will be unpacked into the top level of the `EventDescriptor` document